test/e2e/nodeportlocal_test.go

```diff
@@ -0,0 +1,413 @@
+
+	var nplAnnotations []k8s.NPLAnnotation
+	json.Unmarshal([]byte(nplAnnotationString), &nplAnnotations)
+
+	clientName := randName("test-client-")
+	err = nplTestData.createBusyboxPodOnNode(clientName, node)
+	r.NoError(err, "Error when creating Pod %s", clientName)
+
+	err = nplTestData.podWaitForRunning(defaultTimeout, clientName, testNamespace)
+	r.NoError(err, "Error when waiting for Pod %s to be running", clientName)
+
+	antreaPod, err := nplTestData.getAntreaPodOnNode(node)
+	r.NoError(err, "Error when getting Antrea Agent Pod on Node '%s'", node)
+
+	checkNPLRulesForPod(t, nplTestData, r, nplAnnotations, antreaPod, testPodIP, true)
+	validatePortInRange(t, nplAnnotations, defaultStartPort, defaultEndPort)
+	checkTrafficForNPL(t, nplTestData, r, nplAnnotations, clientName)
+
+	nplTestData.deletePod(testPodName)
+	checkNPLRulesForPod(t, nplTestData, r, nplAnnotations, antreaPod, testPodIP, false)
+}
+
+// TestNPLMultiplePodsAndAgentRestart tests NodePortLocal functionalities after Antrea Agent restarts.
+// - Create multiple Nginx Pods.
+// - Restart Antrea Agent Pods.
+// - Verify Pod Annotation, IPTABLES rules and traffic to test Pod.
+func TestNPLMultiplePodsAgentRestart(t *testing.T) {
+	skipIfNotIPv4Cluster(t)
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+	enableNPLInConfigmap(t, data)
+	r := require.New(t)
+
+	annotation := make(map[string]string)
+	annotation[k8s.NPLEnabledAnnotationKey] = "true"
+	ipFamily := corev1.IPv4Protocol
+	data.createNginxClusterIPServiceWithAnnotations(false, &ipFamily, annotation)
+
+	node := nodeName(0)
+	var testPods []string
+
+	for i := 0; i < 4; i++ {
+		testPodName := randName("test-pod-")
+		testPods = append(testPods, testPodName)
+		err = data.createNginxPod(testPodName, node)
+		r.NoError(err, "Error creating test Pod: %v", err)
+	}
+
+	clientName := randName("test-client-")
+	err = data.createBusyboxPodOnNode(clientName, node)
+	r.NoError(err, "Error when creating Pod %s", clientName)
+
+	err = data.podWaitForRunning(defaultTimeout, clientName, testNamespace)
+	r.NoError(err, "Error when waiting for Pod %s to be running", clientName)
+
+	err = data.restartAntreaAgentPods(defaultTimeout)
+	r.NoError(err, "Error when restarting Antrea Agent Pods")
+
+	antreaPod, err := data.getAntreaPodOnNode(node)
+	r.NoError(err, "Error when getting Antrea Agent Pod on Node '%s'", node)
+
+	for _, testPodName := range testPods {
+		nplAnnotationString, testPodIP := getNPLAnnotation(t, data, r, testPodName)
+		var nplAnnotations []k8s.NPLAnnotation
+		json.Unmarshal([]byte(nplAnnotationString), &nplAnnotations)
+
+		checkNPLRulesForPod(t, data, r, nplAnnotations, antreaPod, testPodIP, true)
+		validatePortInRange(t, nplAnnotations, defaultStartPort, defaultEndPort)
+		checkTrafficForNPL(t, data, r, nplAnnotations, clientName)
+
+		data.deletePod(testPodName)
```

Comment:
this is not technically required and I think you can remove it to speed up the tests: the test namespace is deleted after each test, and that means the Pods should be deleted automatically.